internal/proxy/proxy.go

```diff
@@ -476,60 +520,72 @@ func (p *Proxy) HandleError(w http.ResponseWriter, err error) {
 
 	eb, err := json.Marshal(e)
 	if err != nil {
-		log.Error(err)
+		requestLogEntry.Error(err)
 	}
 	w.Write(eb)
 }
 
-//GHHookStruct a simplified structure to get info from
-//a webhook request
-type GHHookStruct struct {
-	Repository struct {
-		Name     string `json:"name"`
-		FullName string `json:"full_name"`
-		GitURL   string `json:"git_url"`
-		CloneURL string `json:"clone_url"`
-	} `json:"repository"`
+func (p *Proxy) isGitHubRequest(r *http.Request) bool {
+	isGH := false
+	// Is the request coming from Github Webhook?
+	// FIXME - should we only care about push?
+	if ua, exist := r.Header[GHHeader]; exist {
+		isGH = strings.HasPrefix(ua[0], GHAgent)
+	}
+	return isGH
+}
+
+func (p *Proxy) createRequestHash(url string, headers string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(url + headers + fmt.Sprint(time.Now())))
+	return h.Sum32()
 }
 
 //GetUser returns a namespace name based on GitHub repository URL
-func (p *Proxy) GetUser(pl GHHookStruct) (namespace string, err error) {
+func (p *Proxy) GetUser(pl GHHookStruct, requestLogEntry *log.Entry) (string, error) {
 	if n, found := p.TenantCache.Get(pl.Repository.CloneURL); found {
-		log.Info("Cache hit")
-		namespace = n.(string)
-		return
+		namespace := n.(string)
+		requestLogEntry.Infof("Cache hit for repository %s: %s", pl.Repository.CloneURL, namespace)
```

Comment:
Maybe add `ns` field?

Reply:
+1

Reply:
On that note, any update why Kibana is not splitting our JSON?